.github/workflows/build_and_test.yml

```diff
@@ -46,16 +46,33 @@ jobs:
       - name: Run mage integrations unit tests
         run: |
           cd mage_integrations && python3 -m unittest discover mage_integrations.tests
-  build_frontend:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
+
+      # Run Playwright web UI tests only run once per workflow (Python 3.10).
       - name: Set up Node
         uses: actions/setup-node@v3
         with:
           node-version: "14.19.1"
-      - name: Install dependencies
-        run: yarn
-      - name: Build and export
+        if: matrix.python-version == '3.10'
+      - name: Create Mage test project
+        run: |
+          python mage_ai/cli/main.py init test_project &&
+          echo 'help_improve_mage: false' >> test_project/metadata.yaml
+        env:
+          PYTHONPATH: ${{ env.pythonLocation }}:.
+        if: matrix.python-version == '3.10'
+      - name: Build frontend, start server, and run Playwright tests
         run: |
-          cd mage_ai/frontend && yarn install && yarn export
+          yarn install &&
+          yarn export_prod &&
+          yarn playwright install chromium &&
+          yarn playwright test -c playwright.config.ci.ts
+        working-directory: mage_ai/frontend/
+        env:
+          PYTHONPATH: ${{ env.pythonLocation }}:. # Playwright tests on CI depend on the Python web server
+        if: matrix.python-version == '3.10'
+      - uses: actions/upload-artifact@v3
+        if: always()
```

Comment:
Does this step need to be executed for every python version? Seems like we can just run it for 3.10 since running it for the other versions doesn't do anything (this was the output: `No files were found with the provided path: mage_ai/frontend/playwright-report/. No artifacts will be uploaded.`).

Reply:
Everything else looks good, thanks Eric!

Reply:
No, it's true that it should be for 3.10, but this approach uses `always()` which forces this step to be run, even if previous steps failed and the job is cut short. Do you recommend something else?

Reply:
@erictse That seems fine then since running that step is innocuous.